test/contrib/test_socks.py

```diff
@@ -236,7 +236,12 @@ def request_handler(listener):
             handler = handle_socks5_negotiation(sock, negotiate=False)
             addr, port = next(handler)
 
-            self.assertIn(addr, ['127.0.0.1', '::1'])
+            # We're not using assertIn because the tests need to pass under the
+            # old Python 2.6
+            if not addr in ['127.0.0.1', '::1']:
+                raise self.failureException(
+                    "Requested address is %s, 127.0.0.1 or ::1 expected" %
+                    safe_repr(addr))
```

Comment:
There's no need to do this. You can leave the `self.assertIn` statement and then just edit `dummyserver/testcase.py` to change `import unittest` to

```python
import sys

if sys.version_info >= (2, 7):
    import unittest
else:
    import unittest2 as unittest
```

That should resolve the problem.